mu4/domain/notation/internal/msczmetareader.cpp

```diff
@@ -0,0 +1,345 @@
+
+    meta.val.thumbnail = loadThumbnail(&zipReader);
+
+    return meta;
+}
+
+void MsczMetaReader::doReadBox(QXmlStreamReader& xmlReader, RawMeta& meta) const
+{
+    while (xmlReader.readNextStartElement()) {
+        if (xmlReader.name() == "Text") {
+            bool isTitle = false;
+            bool isComposer = false;
+            while (xmlReader.readNextStartElement()) {
+                const QStringRef& tag(xmlReader.name());
+
+                if (tag == "style") {
+                    QString val(xmlReader.readElementText().toLower());
+
+                    if (val == "title" || val == "2") {
+                        isTitle = true;
+                    } else if (val == "composer" || val == "4") {
+                        isComposer = true;
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else if (tag == "text") {
+                    if (isTitle) {
+                        meta.titleStyle = formatFromXml(xmlReader.readElementText(
+                                                            QXmlStreamReader::IncludeChildElements));
+                    } else if (isComposer) {
+                        meta.composerStyle
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else if (tag == "html-data") {
+                    if (isTitle) {
+                        meta.titleStyleHtml
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else if (isComposer) {
+                        meta.composerStyleHtml
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else {
+                    xmlReader.skipCurrentElement();
+                }
+            }
+        } else {
+            xmlReader.skipCurrentElement();
+        }
+    }
+}
+
+MsczMetaReader::RawMeta MsczMetaReader::doReadRawMeta(QXmlStreamReader& xmlReader) const
+{
+    RawMeta meta;
+
+    while (xmlReader.readNextStartElement()) {
+        const QStringRef& tag(xmlReader.name());
+        if (tag == "work-title") {
+            meta.titleTag = xmlReader.readElementText();
+        } else if (tag == "metaTag") {
+            QString name = xmlReader.attributes().value("name").toString();
+            if (name == "workTitle") {
+                meta.titleAttribute = xmlReader.readElementText();
+            } else if (name == "composer") {
+                meta.composerAttribute = xmlReader.readElementText();
+            } else if (name == "arranger") {
+                meta.arranger = xmlReader.readElementText();
```

Comment:
what about lyricist/poet?

Reply:
Thanks. 
Rename poet to lyricist. Added reading lyricist, copyright, translator, subtitle.